eng/common/scripts/SemVer.ps1

```diff
@@ -44,7 +74,7 @@ class SemVer {
         }      
         else
         {
-            throw "Invalid version string: $versionString"
```

Comment:
One side benefit of using a function instead of a class is we can easily return null here for invalid versions and not have to handle exceptions in cases were we are parsing a bunch of versions that may or may not be valid.  I'm considering the use case where I parse all the tags in https://github.com/Azure/azure-sdk/blob/master/eng/scripts/Update-Release-Versions.ps1#L70 as an example. As it stands to use this class I will have to add another method that doesn't throw and just returns $null if invalid.

Reply:
I've added a static method to help with this.